src/diff/patch.js

```diff
@@ -70,11 +69,16 @@ export function patch(
 			nextDomSibling = internal._dom.nextSibling;
 		}
 
-		if (options.diffed) options.diffed(newVNode);
+		if (options.diffed) options.diffed(internal);
+
+		// We successfully rendered this VNode, unset any stored hydration/bailout state:
+		internal._mode = MODE_NONE;
+		// Once we have successfully rendered the new VNode, copy it's ID over
+		internal._original = newVNode._original;
```

Comment:
Once a newVNode has successfully rendered, we should copy it's ID over in case it is memoized and that VNode id is used again.